plugins/node/opentelemetry-instrumentation-mysql2/test/mysql.test.ts

```diff
@@ -1210,13 +1205,160 @@ describe('mysql2', () => {
                   JSON.stringify(res)
                 );
                 done();
-              }
-            );
+              });
+            });
+          });
+        });
+
+        it('should extract data from responseHook - poolCluster', done => {
+          poolCluster.getConnection((err, poolClusterConnection) => {
+            assert.ifError(err);
+            const span = provider.getTracer('default').startSpan('test span');
+            context.with(trace.setSpan(context.active(), span), () => {
+              const sql = 'SELECT 1+1 as solution';
+              poolClusterConnection.query(
+                sql,
+                (err, res: mysqlTypes.RowDataPacket[]) => {
+                  assert.ifError(err);
+                  assert.ok(res);
+                  assert.strictEqual(res[0].solution, 2);
+                  const spans = memoryExporter.getFinishedSpans();
+                  assert.strictEqual(spans.length, 1);
+                  assertSpan(spans[0], sql);
+                  assert.strictEqual(
+                    spans[0].attributes[queryResultAttribute],
+                    JSON.stringify(res)
+                  );
+                  done();
+                }
+              );
+            });
           });
         });
       });
     });
   });
+  describe('promise API', () => {
+    let instrumentation: MySQL2Instrumentation;
+
+    let contextManager: AsyncHooksContextManager;
+    const provider = new BasicTracerProvider();
+    const memoryExporter = new InMemorySpanExporter();
+
+    let connection: mysqlTypesProm.Connection;
+    let rootConnection: mysqlTypesProm.Connection;
+    let mysqlTypesPromReload: typeof mysqlTypesProm;
+
+    before(async function () {
+      // cleanup cache for mysql2
```

Comment:
```suggestion
      // cleanup cache for 'mysql2'
      // in previous iterations these tests would have passed since 'mysql2' was already loaded,
      // but 'mysql2/promise' does not import 'mysql2'. Therefore, if used without ever loading 'mysql2',
      // the relevant code would never get patched.
```